src/vivarium/framework/lookup.py

```diff
@@ -8,92 +8,131 @@
 
 
-class ScalarView(TableView):
-    def __init__(self, value):
-        self.value = value
+class ScalarTable:
+    """A callable that returns a series or dataframe from a single value or list of values with no interpolation.
+
+    Notes
+    -----
+    These should not be created directly. Use the `lookup` method on the builder during setup.
+    """
+    def __init__(self, values, value_columns):
+        if isinstance(values, (list, tuple)):
+            if not value_columns:
+                raise ValueError(f'To invoke scalar view with multiple values, you must supply value_columns')
+            if len(value_columns) != len(values):
+                raise ValueError(f'The number of value columns must match the number of values.'
+                                 f'You supplied values: {values} and value_columns: {value_columns}')
+
+        self._values = values
+        self._value_columns = value_columns
 
     def __call__(self, index):
-        return pd.Series(self.value, index=index)
+        """Returns a series if only one value column, pandas.DataFrame otherwise"""
+        if not isinstance(self._values, (list, tuple)):
+            return pd.Series(self._values, index=index, name=self._value_columns[0] if self._value_columns else None)
+        values = dict(zip(self._value_columns, [pd.Series(v, index=index) for v in self._values]))
+        return pd.DataFrame(values)
 
     def __repr__(self):
-        return "ScalarView(value={})".format(self.value)
+        return "ScalarTable(value(s)={})".format(self._values)
 
 
-class InterpolatedDataManager:
-    """Container for interpolation functions over input data. Interpolation can
-    be turned off on a case by case basis in which case the data will be
-    delegated to MergedTableManager instead.
+class LookupTable:
+    """Container for ScalarTables/InterpolatedTables over input data.
+
+    Notes
+    -----
+    These should not be created directly. Use the `lookup` method on the builder during setup.
+    """
+    def __init__(self, data, population_view, key_columns, parameter_columns, value_columns,
+                 interpolation_order, clock):
+
+        if data is None or (isinstance(data, pd.DataFrame) and data.empty):
+            raise ValueError("Must supply some data")
+
+        # Note datetime catches pandas timestamps
+        if isinstance(data, (Number, datetime, timedelta, list, tuple)):
+            self.table = ScalarTable(data, value_columns)
+
+        elif isinstance(data, pd.DataFrame):
+            if set(key_columns).intersection(set(parameter_columns)):
+                raise ValueError(f'There should be no overlap between key columns: {key_columns} '
+                                 f'and parameter columns: {parameter_columns}.')
+
+            view_columns = sorted((set(key_columns) | set(parameter_columns)) - {'year'})
+            self.table = InterpolatedTable(data, population_view(view_columns), key_columns,
```

Comment:
The table here, on the other hand, should be private.  The `LookupTable` instance is the user's interface into this system.  We don't want them digging around in the stuff underneath.